src/dash-table/components/Table/props.ts

```diff
@@ -25,6 +25,23 @@ export interface IDerivedData {
     indices: Indices;
 }
 
+export interface IViewportOffset {
+    rows: number;
+    columns: number;
+}
+
+export interface IViewportPadding {
+    before: number;
+    after: number;
+}
```

Comment:
Padding extra rows before and after the viewport makes certain behavior easier to keep (e.g. keyboard navigation on the last visible row) -- it can also be used later to optimize virtualization (with a large enough padding it would be possible *not* to re-render and recalculate on all scrolls, only when a certain threshold is met)